libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1066,9 +1080,30 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 		solAssert(false, "Index access only allowed for mappings or arrays.");
 }
 
-void IRGeneratorForStatements::endVisit(IndexRangeAccess const&)
+void IRGeneratorForStatements::endVisit(IndexRangeAccess const& _indexRangeAccess)
 {
-	solUnimplementedAssert(false, "Index range accesses not yet implemented.");
+	Type const& baseType = *_indexRangeAccess.baseExpression().annotation().type;
+
+	if (baseType.category() == Type::Category::Array)
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(baseType);
+		solAssert(_indexRangeAccess.startExpression(), "Start index expression expected.");
+		solAssert(_indexRangeAccess.endExpression(), "End index expression expected.");
+
+		switch (arrayType.location())
+		{
+			case DataLocation::CallData:
+			{
+				IRVariable range{_indexRangeAccess};
+				declare(range);
+				break;
```

Comment:
A good initial implementation for testing might be:
```
IRVariable range{_indexRangeAccess};
declare(range);
assign(range.part("offset"), IRVariable(*_indexRangeAccess.baseExpression()).part("offset"));
assign(range.part("length"), IRVariable(*_indexRangeAccess.baseExpression()).part("length"));
```
That together with the "identity" conversion function should give a good basis for testing and refining.

Reply:
Makes sense, did it now.